src/main.cc

```diff
@@ -224,8 +224,21 @@ static int daemon_main(LiteClient& client, const bpo::variables_map& variables_m
     interval = variables_map["interval"].as<uint64_t>();
   }
 
+  std::vector<Uptane::Target> known_versions;
+  client.storage->loadPrimaryInstallationLog(&known_versions, false);
+
   std::vector<Uptane::Target> installed_versions;
-  client.storage->loadPrimaryInstallationLog(&installed_versions, false);
+  client.storage->loadPrimaryInstallationLog(&installed_versions, true);
+
+  std::vector<Uptane::Target> known_but_not_installed_versions;
+  for (const auto& t : known_versions) {
+    if (installed_versions.end() ==
+        std::find_if(installed_versions.begin(), installed_versions.end(),
+                     [&t](const Uptane::Target& t1) { return t.filename() == t1.filename(); })) {
+      // known but never successfully installed version
+      known_but_not_installed_versions.push_back(t);
+    }
+  }
```

Comment:
you and ricardo seem to have hashed this out, so i won't say much. I wonder if you can push this into a helper function that we could then set up a unit test with?

Reply:
> I wonder if you can push this into a helper function that we could then set up a unit test with?

done.

Reply:
thanks. i'll leave it to you and ricardo to decide now.

Reply:
@ricardosalveti have you had a chance to verify it?

Reply:
Nice, thanks for the tests. Looking good, testing this today and will reply back.